### DISCUSSION TITLE
Escaping \( delimiter when used outside of math expressions?

### DISCUSSION THREAD

**Discussion by oddhack** [KaTeX/KaTeX#3302] [Q&A]
We're using asciidoctor + auto-render extension + offline pre-processing with nodejs to pre-convert math to HTML, and running into an issue where some '\\( ... \\)' markup in text that *isn't* math (a regexp example) is getting treated as a math delimiter and messing up the rendering. The \\( math delimiters are getting generated from somewhere inside the asciidoctor pipeline and aren't directly under our control.

Is there a way to tag regions of the document so the auto-render extension won't run on them? That may be doable via a CSS style on the regexp, for example. Other alternatives get uglier (trying to use funky Unicode to get the offending markup through to the HTML renderer without the auto-render extension seeing it? Actually changing the regexp? Hacking asciidoctor internals?)

**Comment by edemaine** [comment]
Perhaps you can use the `ignoredTags` or `ignoredClasses` options in auto-render? See [this documentation](https://katex.org/docs/autorender.html).

If you want to change the regexes, you could use `[(]` and `[)]` in place of `\(` and `\)`. But maybe that's not an option given that they "aren't directly under [y]our control".

Alternatively, you could change your KaTeX delimiters to ignore `\(` and `\)`, and use something else in their place (`$...$` for example, unless you have regexes with `$`s in them...)

The "proper" way to fix this would be to embed KaTeX into asciidoctor so that they're processed together and proper escaping can happen.  Maybe look at some plugins like https://github.com/jirutka/asciidoctor-katex

**Reply by oddhack** [reply]
> Perhaps you can use the `ignoredTags` or `ignoredClasses` options in auto-render? See [this documentation](https://katex.org/docs/autorender.html).
> 
> If you want to change the regexes, you could use `[(]` and `[)]` in place of `\(` and `\)`. But maybe that's not an option given that they "aren't directly under [y]our control".
> 
> Alternatively, you could change your KaTeX delimiters to ignore `\(` and `\)`, and use something else in their place (`$...$` for example, unless you have regexes with `$`s in them...)
> 
> The "proper" way to fix this would be to embed KaTeX into asciidoctor so that they're processed together and proper escaping can happen. Maybe look at some plugins like https://github.com/jirutka/asciidoctor-katex

Thanks. Eventually I realized moving to POSIX EREs allowed using non-escaped () and given that it's one line in a 3,000 page document, that was sufficient for the time being.

I'd love to see katex more tightly integrated with asciidoctor eventually. Today we have to use a Ruby extension to asciidoctor to replace the MathJax \<script> tags with equivalent KaTeX font/script links, then do the nodejs postprocess to avoid runtime rendering and the major performance hit that imposes on our specs.
